api/maven-api-settings/src/main/mdo/settings.mdo

```diff
@@ -535,6 +539,62 @@
           <type>String</type>
         </field>
       </fields>
+      <codeSegments>
+        <codeSegment>
+          <version>1.0.0/1.3.0</version>
+          <code>
+            <![CDATA[
+    public boolean isActive() {
+        return (getActiveString() != null) ? Boolean.parseBoolean(getActiveString()) : true;
+    }
+
+    public void setActive(boolean active) {
+        setActiveString(String.valueOf(active));
+    }
+
+    public int getPort() {
+        return (getPortString() != null) ? Integer.parseInt(getPortString()) : 8080;
+    }
+
+    public void setPort(int port) {
+        setPortString(String.valueOf(port));
```

Comment:
This allows any port value, including negative numbers and other ints that aren't legal port values. An IllegalArgumentException might helpful.


[IllegalArgumentException](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html) - if the port parameter is outside the specified range of valid port values, which is between 0 and 65535, inclusive.

Reply:
We don't do any validation in the model afaik, it is done later using custom components such as `SettingsValidator` or `ModelValidator`.  Although the goal of those methods is to provide compatibility with the ones that are currently automatically generated which look like:
```
    /**
     * Get the proxy port.
     * 
     * @return int
     */
    public int getPort()
    {
        return this.port;
    } //-- int getPort()
```
and
```
    /**
     * Set the proxy port.
     * 
     * @param port a port object.
     */
    public void setPort( int port )
    {
        this.port = port;
    } //-- void setPort( int )
```

Reply:
Interesting. Why don't we? In general, the ealrier one can detect an error such as a negative port the better.